include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -762,8 +769,10 @@ V [ 0,4,8...60   1,5...61     2,6...62     3,7...63    ]   [ 128,132...188  129,
 
   let parameters = (
     ins
-    "unsigned":$nonKDim,
+    "unsigned": $versionMajor,
+    "unsigned": $versionMinor,
```

Comment:
Maybe add type information here?

I think `A data type` + `B data type` + instr shape will be enough to identify instruction we need to emit.

Plus I would like to use this information to move tensor casts around dot operands from python code(semantic.py) to C++ pass

Reply:
I don't think we want to include elem type info here
- The mma layout does not have it. This code is upstream now and in the future we may want to merge mfma and mma.
- When we pick what mfma instruction to emit, we need {elemtype_a, elemtype_b, matrixVersion} as the key. So you should have element type information before mfma layout is constructed. And you can use that information to cast tensors.
- The layout stuff is mained used for data distribution. It's more about how many elements each thread has. So element type info is not needed. We do need the `matrixVersion` info, since it affects the element distribution of dotOperand layouts.